tests/drivers/build_all/eeprom/app.overlay

```diff
@@ -51,6 +51,16 @@
 				wp-gpios = <&test_gpio 0 0>;
 				/* read-only; */
 			};
+
+			test_i2c_24xx01: 24xx01@2 {
+				compatible = "microchip,24xx01-eeprom";
+				reg = <0x2>;
+				size = <128>;
+				pagesize = <8>;
+				write-cycle-time = <5>;
+				wp-gpios = <&test_gpio 0 0>;
+				/* read-only; */
+			};
```

Comment:
From what I can tell from the data sheet, the Microchip 24xx01 series of EEPROMs should already be supported by the https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/eeprom/eeprom_at2x.c driver. Please try a devicetree entry like the following for testing:

        eeprom@50 {
                compatible = "atmel,at24";
                reg = <0x50>;
                size = <128>;
                pagesize = <8>;
                address-width = <8>;
                timeout = <5>;
        };

Reply:
Hi, thank you for your suggestion. The atmel,at24 compatible does indeed seems to work for the Microchip 24xx01. Shouldn't this compatible be made more generic since not only Atmel eeproms are supported?

Reply:
Atmel launched the initial AT24xx I2C EEPROMs using this protocol, so the bindings are historically bound the their vendor ID. The binding does mention that it supports Atmel AT24 "and compatible" EEPROMs: https://docs.zephyrproject.org/latest/build/dts/api/bindings/mtd/atmel,at24.html

The binding matches the Linux kernel devicetree bindings for the same class of EEPROMs: https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/eeprom/at24.yaml

The same goes for the Atmel AT25 (and compatible) SPI EEPROMs, by the way: https://docs.zephyrproject.org/latest/build/dts/api/bindings/mtd/atmel,at25.html and https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/eeprom/at25.yaml

You can list multiple compatibles in your devicetree for these EEPROMs, as long as the least-specific compatible string is "atmel,at24".

Reply:
Ok, thank you

Reply:
@allemanm Can this PR be closed?

Reply:
Yes